app/src/main/java/com/infomaniak/mail/utils/extensions/Extensions.kt

```diff
@@ -595,6 +588,20 @@ fun Fragment.bindAlertToViewLifecycle(alertDialog: BaseAlertDialog) {
     alertDialog.bindAlertToLifecycle(viewLifecycleOwner)
 }
 
+/**
+ * Asynchronously validate folder name locally
+ * @return error string, otherwise null
+ */
+private val invalidCharactersRegex by lazy { Regex("[/'\"]") }
+fun Context.checkForFolderCreationErrors(folderName: CharSequence, folderController: FolderController): String? {
+    return when {
+        folderName.length > 255 -> getString(R.string.errorNewFolderNameTooLong)
+        folderName.contains(invalidCharactersRegex) -> getString(R.string.errorNewFolderInvalidCharacter)
+        folderController.getRootFolder(folderName.toString()) != null -> getString(R.string.errorNewFolderAlreadyExists)
```

Comment:
Since you're returning a string error, maybe a suitable name cloud be `getFolderCreationError` ?